packages/interactive_media_ads/lib/src/android/android_ad_display_container.dart

```diff
@@ -0,0 +1,241 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/widgets.dart';
+import 'package:meta/meta.dart';
+
+import '../platform_interface/platform_interface.dart';
+import 'android_view_widget.dart';
+import 'interactive_media_ads.g.dart' as ima;
+import 'interactive_media_ads_proxy.dart';
+import 'platform_views_service_proxy.dart';
+
+/// Android implementation of [PlatformAdDisplayContainerCreationParams].
+final class AndroidAdDisplayContainerCreationParams
+    extends PlatformAdDisplayContainerCreationParams {
+  /// Constructs a [AndroidAdDisplayContainerCreationParams].
+  const AndroidAdDisplayContainerCreationParams({
+    super.key,
+    required super.onContainerAdded,
+    @visibleForTesting InteractiveMediaAdsProxy? imaProxy,
+    @visibleForTesting PlatformViewsServiceProxy? platformViewsProxy,
+  })  : _imaProxy = imaProxy ?? const InteractiveMediaAdsProxy(),
+        _platformViewsProxy =
+            platformViewsProxy ?? const PlatformViewsServiceProxy(),
+        super();
+
+  /// Creates a [AndroidAdDisplayContainerCreationParams] from an instance of
+  /// [PlatformAdDisplayContainerCreationParams].
+  factory AndroidAdDisplayContainerCreationParams.fromPlatformAdDisplayContainerCreationParams(
+    PlatformAdDisplayContainerCreationParams params, {
+    @visibleForTesting InteractiveMediaAdsProxy? imaProxy,
+    @visibleForTesting PlatformViewsServiceProxy? platformViewsProxy,
+  }) {
+    return AndroidAdDisplayContainerCreationParams(
+      key: params.key,
+      onContainerAdded: params.onContainerAdded,
+      imaProxy: imaProxy,
+      platformViewsProxy: platformViewsProxy,
+    );
+  }
+
+  final InteractiveMediaAdsProxy _imaProxy;
+  final PlatformViewsServiceProxy _platformViewsProxy;
+}
+
+/// Android implementation of [PlatformAdDisplayContainer].
+base class AndroidAdDisplayContainer extends PlatformAdDisplayContainer {
+  /// Constructs an [AndroidAdDisplayContainer].
+  AndroidAdDisplayContainer(super.params) : super.implementation() {
+    final WeakReference<AndroidAdDisplayContainer> weakThis =
+        WeakReference<AndroidAdDisplayContainer>(this);
+    _videoView = _setUpVideoView(weakThis);
+    _frameLayout.addView(_videoView);
+    _videoAdPlayer = _setUpVideoAdPlayer(weakThis);
+  }
+
+  // The duration between each update to the IMA SDK of the progress of the
+  // currently playing ad.
+  static const int _progressPollingMs = 250;
```

Comment:
Could you explain in the comment where/how this number was derived? I.e., is it arbitrary, based on best practice docs, other?

Reply:
It was the same value as the one in the android example: https://developers.google.com/interactive-media-ads/docs/sdks/android/client-side#6.-create-the-videoadplayeradapter-class. It doesn't give an explanation, but I added a comment to explain where the value is from.